### DISCUSSION TITLE
Implement button to force bootloader mode

### DISCUSSION THREAD

**Discussion by ccrome** [adafruit/tinyuf2#293] [General]
It looks as if there was a thought of using a button to force bootloader mode, but it was never implemented, is that right?

For example, https://github.com/adafruit/tinyuf2/blob/master/ports/mimxrt10xx/boards/imxrt1060_evk/board.h, there are defines for BUTTON_PINMUX, PORT, etc, but they are never used in boards.c, and further, there does not seem to be any attempt to read a button inside check_dfu_mode.

I'd like to have that functionality -- am I the only one? 

The behavior I'd like to see in check_dfu_mode is something like this:
```
#if TINYUF2_DFU_BUTTON
if (board_button_read())
    return true;
#endif
```

This is particularly useful on the teensy, since there is no access to the POR line.

Does it make sense to open a PR for that, or is that just something I should keep in my own repo?

**Comment by hathach** [comment]
Yeah, there is an intention to do so, however, since double reset works so well for us so far. I haven't bothered doing so. An PR is welcome, I will be happy to review.

**Reply by ccrome** [reply]
https://github.com/adafruit/tinyuf2/pull/294
